src/test/java/com/crowd/controller/UserApiControllerChangePasswordMethodTest.java

```diff
@@ -0,0 +1,103 @@
+package com.crowd.controller;
+
+import com.crowd.bean.RegistrationFields;
+import com.crowd.bean.user.ChangePasswordRequest;
+import com.crowd.bean.user.ChangePasswordResponse;
+import com.crowd.bean.user.UserBean;
+import com.crowd.config.TestHibernateConfig;
+import com.crowd.controller.api.UserApiController;
+import com.crowd.service.UserService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.util.Asserts;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.AnnotationConfigWebContextLoader;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import javax.transaction.Transactional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.model;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {TestHibernateConfig.class}, loader = AnnotationConfigWebContextLoader.class)
+@WebAppConfiguration
+@Transactional
+public class UserApiControllerChangePasswordMethodTest {
```

Comment:
When you test service with database in use you may replace real services with mocks in controller test.